core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -198,9 +205,11 @@ public ChannelImpl shutdown() {
    *
    * <p>NOT YET IMPLEMENTED. This method currently behaves identically to shutdown().
    */
-  // TODO(ejona86): cancel preexisting calls.
   public ChannelImpl shutdownNow() {
     shutdown();
+    for (ClientCall<?, ?> call : activeCalls) {
+      call.cancel();
```

Comment:
If it does, then it has a bug and probably is old; before error handling was really defined. Look at the implementation: cancel is not thread-safe. Read the API: cancel is not defined as thread-safe.